ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2018 Hyperwallet Systems Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.hyperwallet.android.ui.view.widget;
+
+import android.text.TextUtils;
+import android.text.format.DateFormat;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Class is used for manage and convert date {@link DateWidget}
+ */
+public final class DateUtil {
```

Comment:
In this class we should not handle any parsing exception. if the data comes from the server, it should be correct. If we do run into a parse exception, there's no difference between catching it and throwing DateParseException because both would be handled the same way. So in isServerDateValid method, we just validate if the date is actually present (not empty) if it is not empty, do not handle parse exception

Reply:
Done